src/Components/CreateImageWizardV2/steps/Repositories/Repositories.tsx

```diff
@@ -40,224 +43,173 @@ import {
-export const convertSchemaToIBPayloadRepo = (
-  repo: ApiRepositoryResponseRead
-) => {
-  const imageBuilderRepo: Repository = {
-    baseurl: repo.url,
-    rhsm: false,
-    check_gpg: false,
-  };
-  // only include the flag if enabled
-  if (repo.module_hotfixes) {
-    imageBuilderRepo.module_hotfixes = repo.module_hotfixes;
-  }
-  if (repo.gpg_key) {
-    imageBuilderRepo.gpgkey = repo.gpg_key;
-    imageBuilderRepo.check_gpg = true;
-    imageBuilderRepo.check_repo_gpg = repo.metadata_verification;
-  }
-
-  return imageBuilderRepo;
-};
+import useDebounce from '../../../../Utilities/useDebounce';
 
-const Repositories = () => {
+const Repositories = ({ refresh }: { refresh: () => void }) => {
   const dispatch = useAppDispatch();
   const wizardMode = useAppSelector(selectWizardMode);
   const arch = useAppSelector(selectArchitecture);
   const distribution = useAppSelector(selectDistribution);
   const version = releaseToVersion(distribution);
-  const repositoriesList = useAppSelector(selectCustomRepositories);
+  const customRepositories = useAppSelector(selectCustomRepositories);
+  const packages = useAppSelector(selectPackages);
+  const payloadRepositories = useAppSelector(selectPayloadRepositories);
   const recommendedRepos = useAppSelector(selectRecommendedRepositories);
 
-  const [isRepoRemovalModalOpen, setIsRepoRemovalModalOpen] = useState(false);
-  const [repoToRemove, setRepoToRemove] = useState('');
   const [filterValue, setFilterValue] = useState('');
   const [perPage, setPerPage] = useState(10);
   const [page, setPage] = useState(1);
-  const [toggleSelected, setToggleSelected] = useState('toggle-group-all');
-  const [selected, setSelected] = useState(
-    repositoriesList ? repositoriesList.flatMap((repo) => repo.baseurl) : []
+  const [toggleSelected, setToggleSelected] = useState<
+    'toggle-group-all' | 'toggle-group-selected'
+  >('toggle-group-all');
+
+  const debouncedFilterValue = useDebounce(filterValue);
+
+  const selected = useMemo(
+    () =>
+      new Set(
+        [
+          ...customRepositories.map(({ baseurl }) => baseurl || []).flat(1),
+          ...(payloadRepositories.map(({ baseurl }) => baseurl) || []),
+          ...(recommendedRepos.map(({ url }) => url) || []),
+        ].filter((url) => !!url) as string[]
+      ),
+    [customRepositories, payloadRepositories, recommendedRepos]
   );
 
-  const firstRequest = useListRepositoriesQuery(
+  useEffect(() => {
+    if (toggleSelected === 'toggle-group-selected' && !selected.size) {
+      setToggleSelected('toggle-group-all');
+    }
+  }, [selected, toggleSelected]);
+
+  const {
+    data: { data: contentList = [], meta: { count } = { count: 0 } } = {},
+    isError,
+    isFetching,
+    isLoading,
+  } = useListRepositoriesQuery(
     {
       availableForArch: arch,
       availableForVersion: version,
       contentType: 'rpm',
       origin: 'external',
-      limit: 100,
+      limit: perPage,
```

Comment:
Previous double fetching method is not very ideal, but I wonder if we want to fetch based on perPage limit only as any changes cause more loading states. Will return to this in the main comment.

The pagination doesn't seem to work:
![broken_pagination](https://github.com/osbuild/image-builder-frontend/assets/49452678/0954a8e7-826a-42c2-981a-cf73af3d16e3)

Reply:
Probably not caused by this PR: https://github.com/osbuild/image-builder-frontend/issues/2084 :)

Reply:
I believe those are two separate issues as currently the pagination on the Repositories step seems to work. The proposed changes in this PR would break it.

Reply:
Ah sorry! I've gotten confused, you're right, those are separate issues!

Reply:
No worries :)

Reply:
> Ah sorry! I've gotten confused, you're right, those are separate issues!

Actually same issue, I keep forgetting both the page/perPage are needed references for useMemo/dependency arrays.